static/scripts/dashboard/dashboardCtrl.js

```diff
@@ -848,9 +848,24 @@
                   systemIntegration.data
                 );
                 $scope.vm.installForm[sysIntName][masterName].isEnabled = true;
-                if (sysIntName === 'auth')
-                  $scope.vm.installForm[sysIntName][masterName].callbackUrl =
-                    systemIntegration.data.wwwUrl + '/' + sysIntName +'/' + systemIntegration.id + '/identify';
+                if (sysIntName === 'auth') {
+                  if (masterName === 'bitbucketKeys') {
```

Comment:
This code will be a lot easier to read and maintain if you use a map instead of the if/else's.

````javascript
var providerAuthNames = {
  'githubEnterprise': 'ghe',
  ...
  ...
}
...
...
...
var providerAuthName = providerAuthNames[masterName];
$scope.vm.whatever = util.format('%s/auth/%s/%s/identify',
  wwwUrl, providerAuthName, systemIntegration.id);
````